oberon-chat/chat-server
# Server side logic to archive rooms

## Why:
We want to be able to change the state of rooms from active to archived while leaving open the possibly more room states in the future.

## What:
This PR is intended to offer the server side functions in order to update a room state and save it back to the db in use via Ecto.

## Steps To Verify PR:
- [x] Pull branch for PR down locally and complete chat-server setup instructions detailed in [README.md](https://github.com/oberon-chat/chat-server/blob/master/README.md)
- [x] Open a terminal window and run start command located in `bin/start`.
- [x] Inside IEX console run commands and observer results:

```
1. iex> ChatServer.Schema.Room.all
2. Copy room id from any of the rooms listed that are active.
3. iex> room = ChatServer.Schema.Room.get("<ROOM ID>")
4. iex> ChatServer.Schema.User.all
5. Copy user id from any of the users listed to use for this test.
6. iex> user = ChatServer.Schema.User.get("<USER ID>")
7. iex> ChatServer.UpdateState.call(room, user, "archived")
8. Observe output showing room state is now "archived"
9. iex> ChatServer.Schema.Room.all
10. Observe newly polled output from db is also showing room state is now set to "archived".
```
